![](https://i.imgur.com/JsMvGnP.png)


## Exercícios de manipulação de dados - Parte 1

Neste Jupyter notebook você irá resolver uma exercícios utilizando a biblioteca Pandas.\
Todos os datasets utilizados nos exercícios estão salvos na pasta *datasets*.\
Todo o seu código deve ser executado neste Jupyter Notebook. Por fim, se desejar, revise as respostas com o seu mentor.

In [18]:
import pandas as pd

#### Passo 1. Importando os dados

Carregue os dados salvos no arquivo ***datasets/users_dataset.csv***.\
Esse arquivo possui um conjunto de dados de trabalhadores com 5 colunas separadas pelo símbolo "|" (pipe) e 943 linhas.

Dica: não se esqueça do argumento *sep* quando importar os dados.

In [19]:
url = 'https://raw.githubusercontent.com/awarischool/data-manipulation-exercises/master/datasets/users_dataset.csv'
df = pd.read_csv(url, sep='|')

#### Passo 2. Mostre as 15 primeiras linhas do dataset.

*Dica: utilize a função head do DataFrame*

In [20]:
df.head(15)

user_id  age gender  occupation zip_code
0        1   24      M  technician    85711
1        2   53      F       other    94043
2        3   23      M      writer    32067
3        4   24      M  technician    43537
4        5   33      F       other    15213

#### Passo 3. Mostre as 10 últimas linhas do dataset

In [21]:
df.tail(10)

user_id  age gender     occupation zip_code
933      934   61      M       engineer    22902
934      935   42      M         doctor    66221
935      936   24      M          other    32789
936      937   48      M       educator    98072
937      938   38      F     technician    55038
938      939   26      F        student    33319
939      940   32      M  administrator    02215
940      941   20      M        student    97229
941      942   48      F      librarian    78209
942      943   22      M        student    77841

#### Passo 4. Qual o número de linhas e colunas do DataFrame?

In [22]:
df.shape

(943, 5)

O DataFrame possui **943** linhas e **5** colunas

#### Passo 5. Mostre o nome de todas as colunas.

In [23]:
df.columns

Index(['user_id', 'age', 'gender', 'occupation', 'zip_code'], dtype='object')

#### Passo 6. Qual o tipo de dado de cada columa?

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 943 entries, 0 to 942
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   user_id     943 non-null    int64 
 1   age         943 non-null    int64 
 2   gender      943 non-null    object
 3   occupation  943 non-null    object
 4   zip_code    943 non-null    object
dtypes: int64(2), object(3)
memory usage: 37.0+ KB


#### Passo 7. Mostre os dados da coluna *occupation*.

In [34]:
df['occupation']

0         technician
1              other
2             writer
3         technician
4              other
           ...      
938          student
939    administrator
940          student
941        librarian
942          student
Name: occupation, Length: 943, dtype: object

#### Passo 8. Quantas ocupações diferentes existem neste dataset?

In [51]:
len(df['occupation'].unique())

21

Existem **21** diferentes ocupaçoes no dataset

#### Passo 9. Qual a ocupação mais frequente?

In [56]:
df['occupation'].value_counts()

student          196
other            105
educator          95
administrator     79
engineer          67
programmer        66
librarian         51
writer            45
executive         32
scientist         31
artist            28
technician        27
marketing         26
entertainment     18
healthcare        16
retired           14
lawyer            12
salesman          12
none               9
homemaker          7
doctor             7
Name: occupation, dtype: int64

Com **196** indivíduos, os estudantes estão presentes em maior número no dataset

#### Passo 10. Qual a idade média dos usuários?

In [58]:
df['age'].mean()

34.05196182396607

#### Passo 11. Utilize o método describe() para obter diversas informações a respeito do DataFrame.

In [60]:
df.describe()

user_id         age
count  943.000000  943.000000
mean   472.000000   34.051962
std    272.364951   12.192740
min      1.000000    7.000000
25%    236.500000   25.000000
50%    472.000000   31.000000
75%    707.500000   43.000000
max    943.000000   73.000000

**Parabéns! Você chegou ao fim!**